### PR TITLE
runtime(filetype): Modula-2 files with priority not detected

### PR DESCRIPTION
**Pull request by dkearns**
Problem:  Modula-2 files with a specified priority are not detected.
Solution: Match the priority syntax in module header lines when
          performing heuristic content detection.

### CODE REVIEW

File: runtime/autoload/dist/ft.vim [hunk 1]
```diff
@@ -1293,4 +1293,4 @@ export def FTvba()
 enddef
 
 # Uncomment this line to check for compilation errors early
-defcompile
+# defcompile
```
Comment (k-takata) [inline runtime/autoload/dist/ft.vim:1296]: Is this intentional?

Comment (dkearns) [reply runtime/autoload/dist/ft.vim:1296]: Yes, it was accidentally left enabled in 68a8947.  I've added a comment to the commit message.

Thanks for paying attention.

**Comment by chrisbra** [comment]
thanks!

**Review by KidiIT — COMMENTED** [review]
All reviewed and approved🫶❤️❤️

**Pull request merged**